client/sites-dashboard/components/sites-table.tsx

```diff
@@ -15,42 +16,85 @@ interface SitesTableProps {
 const Table = styled.table`
 	border-collapse: collapse;
 	table-layout: fixed;
-	@media only screen and ( max-width: 781px ) {
-		.sites-table__mobile-hidden {
-			display: none;
-		}
-	}
+	position: relative;
 `;
 
+const THead = styled.thead< { top: number } >( ( { top } ) => ( {
+	'@media only screen and ( max-width: 781px )': {
+		display: 'none',
+	},
+
+	position: 'sticky',
+	zIndex: 3,
+	top: `${ top }px`,
+
+	background: '#fdfdfd',
+} ) );
+
 const Row = styled.tr`
 	line-height: 2em;
 	border-bottom: 1px solid #eee;
+
 	th {
 		padding-top: 12px;
 		padding-bottom: 12px;
 		vertical-align: middle;
 		font-size: 14px;
 		line-height: 20px;
 		letter-spacing: -0.24px;
-		font-weight: normal;
+		font-weight: 500;
```

Comment:
@SaxonF had intentionally set the `font-weight` to `normal` in #65871 (https://github.com/Automattic/wp-calypso/pull/65871/files#diff-9a76834adc5c82f0271bab5f707cc9d206758029f2c016192d06bfba7d2e0cc1R30)